mvicore-diff/src/main/java/com/badoo/mvicore/ModelWatcher.kt

```diff
@@ -0,0 +1,52 @@
+package com.badoo.mvicore
+
+class ModelWatcher<T> private constructor(
+    private val watchers: List<Watcher<T, Any?>>
+) {
+    private var state: T? = null
```

Comment:
I think it would be better to rename `state` to `model`. Because `ModelWatcher` is supposed to work with `model`. Otherwise you can rename `ModelWatcher` to `StateWatcher`. 

```suggestion
    private var model: T? = null
```